src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1144,3 +1165,21 @@ def derive_column_states_from_subquery(
             state_dict=column_states,
         )
     return column_states
+
+
+def derive_data_generator_exp(expressions: Optional[List["Expression"]]) -> bool:
+    if expressions is None:
+        return False
+    for exp in expressions:
+        if isinstance(exp, WindowExpression):
+            return True
+        if isinstance(exp, FunctionExpression) and (
+            exp.is_data_generator
+            or exp.name.lower()
+            in ("normal", "zipf", "uniform", "seq1", "seq2", "seq4", "seq8")
```

Comment:
Let's make this a static constant. Also, this is a bit risky when Snowflake introduces new functions. Is there a process to update this list?

Reply:
There may be also metadata columns relevant for this here from our use case.

Reply:
Agree with making them a constant. Metadata columns have to be used along with window, right? Then it should already be handled above.

Reply:
@sfc-gh-sfan Great question. I created a JIRA to follow up. The server side has metadata to identify whether a function is sequence dependent data generator. We need to find a way to retrieve it when a session is created then we can use it. For now I have to use this static list.

Reply:
@sfc-gh-lspiegelberg I merged this fix to a snowpandas branch. All tests except sql counter tests passed.